common/thorhelper/roxiehelper.cpp

```diff
@@ -1107,9 +1102,7 @@ class CHeapSortAlgorithm : public CSortAlgorithm
         eof = false;
         if (inputAlreadySorted)
         {
-            while (sorted.isItem(curIndex))
-                ReleaseRoxieRow(sorted.item(curIndex++));
-            sorted.kill();
+            roxiemem::ReleaseRoxieRows(sorted, curIndex);
```

Comment:
I think in all of these cases it doesn't matter since sorted is killed.  But I agree it would be clearer if the index was also reset at the same time.